django_ipam/api/generics.py

```diff
@@ -49,3 +52,30 @@ def post(self, request, *args, **kwargs):
             headers = self.get_success_headers(serializer.data)
             return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
         return Response(None)
+
+
+class BaseImportSubnetView(CreateAPIView):
+    serializer_class = ImportSubnetSerializer
+    authentication_classes = (SessionAuthentication, BasicAuthentication)
+    permission_classes = (DjangoModelPermissions,)
+
+    def post(self, request, *args, **kwargs):
+        file = request.FILES['csvfile']
+        if not file.name.endswith(('.csv', '.xls', '.xlsx')):
+            return Response('File type not supported.')
+        if not self.subnet_model.import_csv(self, file):
+            return Response('Import data error.')
```

Comment:
same here, you should display a bit more info to the user about what went wrong, I explained how to do this below.

Eg:

```python
try:
    self.subnet_model.import_csv(self, file)
catch CsvImportError as e:
    return Response({'error': e.message, status=400})
```